contracts/CreditGuild.sol

```diff
@@ -0,0 +1,105 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@unioncredit/v1-sdk/contracts/BaseUnionMember.sol";
+
+/*
+credit guild
+*/
+contract CreditGuild is ERC721, BaseUnionMember { 
+// contract CreditGuild is ERC721, BaseUnionMember { // replace with the nouns one. copy it over and change stuff.
+
+  uint256 public id;
+  uint256 public membershipFee;
+  address[] public initialMembers;
+  ERC20 public erc20;
+  // IUserManager public userManager;
+  // ask gerald about making nft non transferrable
+
+  // mapping of existing member to new member to endorsed
+  mapping(address => mapping(address => bool)) public endorsements;
```

Comment:
We can remove this mapping or endorements, we can also remove the endorsed mapping.